src/testing/mockNostojs.ts

```diff
@@ -2,14 +2,22 @@ import { API, nostojs, NostojsCallback } from "../client/nosto"
 
 let originalNostojs: nostojs
 
+type MockMember<T> = T extends (...args: never[]) => unknown
+  ? (...args: Parameters<T>) => Partial<ReturnType<T>>
+  : T extends object
+    ? { [K in keyof T]?: Partial<T[K]> }
+    : T
```

Comment:
Is it deep partial? Looks like it.
I think it needs a better name or a comment explaining the purpose, very hard to understand it on a glance.

Reply:
it is a DeepPartial with capped depth
I'll add a comment